src/server/game/Entities/Player/Player.cpp

```diff
@@ -19309,27 +19309,33 @@ void Player::_LoadMail()
     }
 
     //This should in theory always be < 100
-    for (PlayerMails::iterator itr = GetMailBegin(); itr != GetMailEnd();)
+    for (PlayerMails::iterator itr = m_mailCache.begin(); itr != m_mailCache.end();)
     {
-        Mail* m = *itr;
-        m_mailCache.erase(itr);
-        if(m)
-            delete m;
-        itr = GetMailBegin();
+        Mail* mail = *itr;
+        itr = m_mailCache.erase(itr);
+
+        if (mail)
+            delete mail;
     }
 
     // Delete mailed items aswell
     // Created again below in Player::_LoadMailedItems
     for (ItemMap::iterator iter = mMitems.begin(); iter != mMitems.end(); ++iter)
         delete iter->second;
 
+    std::set<uint32> pendingAuctions;
+    std::unordered_map<uint32, Mail*> pendingAuctionMails;
+
     mMitems.clear();
 
     //Now load the new ones
     m_mailCache.clear();
+
+    SQLTransaction pendingAuctionsTrans = CharacterDatabase.BeginTransaction();
```

Comment:
same here, you should be able to use `auto`

Reply:
No - stop using `auto` with no reason.

Reply:
the reason is to infer the type, this can have different advantages, one of this is to change only one line of code if you need to change the return type of the function

here a real example in azerothcore